rune/libenclave/internal/runtime/pal/skeleton/sgxsign.c

```diff
@@ -257,7 +260,13 @@ static bool measure_encl(const char *path, uint8_t *mrenclave)
 		goto out;
 	}
 
-	if (!mrenclave_ecreate(ctx, sb.st_size))
+	fseek(file, 16L, SEEK_SET);
+	rc = fwrite(&sb.st_size, 1, 8, file);
```

Comment:
Don't **DO** this. You are modifying the original binary file. In order to change the ossa filed, you can refer to encl_data_map which maps the file content into memory. So you can modify the memory copy without touching the binary file.